stl/inc/span

```diff
@@ -0,0 +1,778 @@
+        return reverse_iterator{end()};
+    }
+
+    _NODISCARD constexpr reverse_iterator rend() noexcept {
+        return reverse_iterator{begin()};
+    }
+
+    _NODISCARD constexpr const_reverse_iterator rbegin() const noexcept {
+        return const_reverse_iterator{end()};
+    }
+
+    _NODISCARD constexpr const_reverse_iterator rend() const noexcept {
+        return const_reverse_iterator{begin()};
+    }
+
+    _NODISCARD constexpr const_reverse_iterator crbegin() const noexcept {
+        return rbegin();
+    }
+
+    _NODISCARD constexpr const_reverse_iterator crend() const noexcept {
+        return rend();
+    }
+
+    _NODISCARD constexpr pointer _Unchecked_begin() noexcept {
+        return _Mydata;
+    }
+
+    _NODISCARD constexpr const_pointer _Unchecked_begin() const noexcept {
+        return _Mydata;
+    }
+
+    _NODISCARD constexpr pointer _Unchecked_end() noexcept {
+        return _Mydata + size();
+    }
+
+    _NODISCARD constexpr const_pointer _Unchecked_end() const noexcept {
+        return _Mydata + size();
+    }
+
+private:
+    using _Mybase = _Span_extent_type<_Extent>;
+
+    pointer _Mydata{nullptr};
+};
+
+// DEDUCTION GUIDES
+template <class _Ty, size_t _Extent>
+span(_Ty (&)[_Extent])->span<_Ty, _Extent>;
+
+template <class _Ty, size_t _Extent>
+span(array<_Ty, _Extent>&)->span<_Ty, _Extent>;
+
+template <class _Ty, size_t _Extent>
+span(const array<_Ty, _Extent>&)->span<const _Ty, _Extent>;
+
+#ifdef __cpp_lib_concepts
+
+template <contiguous_iterator _It, class _End>
+span(_It, _End)->span<remove_reference_t<iter_reference_t<_It>>>;
+
+template <_RANGES contiguous_range _Range>
+span(_Range &&)->span<remove_reference_t<_RANGES range_reference_t<_Range>>>;
+
+#else // ^^^ __cpp_lib_concepts / !__cpp_lib_concepts vvv
+
+template <class _Range>
+span(_Range&)->span<typename _Range::value_type>;
+
+template <class _Range>
+span(const _Range&)->span<const typename _Range::value_type>;
+
+#endif // !__cpp_lib_concepts
+
+#if _HAS_STD_BYTE
+// [span.objectrep] Views of object representation
+// FUNCTION TEMPLATE as_bytes
+template <class _Ty, size_t _Extent>
+_NODISCARD auto as_bytes(span<_Ty, _Extent> _Sp) noexcept
+    -> span<const byte, _Extent == dynamic_extent ? dynamic_extent : sizeof(_Ty) * _Extent> {
+    return {reinterpret_cast<const byte*>(_Sp.data()), _Sp.size_bytes()};
+}
+
+// FUNCTION TEMPLATE as_writable_bytes
+template <class _Ty, size_t _Extent, enable_if_t<!is_const_v<_Ty>, int> = 0>
+_NODISCARD auto as_writable_bytes(span<_Ty, _Extent> _Sp) noexcept
+    -> span<byte, _Extent == dynamic_extent ? dynamic_extent : sizeof(_Ty) * _Extent> {
+    return {reinterpret_cast<byte*>(_Sp.data()), _Sp.size_bytes()};
+}
+#endif // _HAS_STD_BYTE
+
+// [span.tuple] Tuple interface
+template <class _Ty, size_t _Extent>
+struct tuple_size<span<_Ty, _Extent>> : integral_constant<size_t, _Extent> {
+    static_assert(dynamic_extent != _Extent, "std::span<T, dynamic_extent> is not tuple-like");

```

Comment:
This `static_assert` is no longer necessary; the "not defined" partial specialization `tuple_size<span<_Ty, dynamic_extent>>` immediately below will absorb this case.

Reply:
Done